corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -0,0 +1,146 @@
+hqDefine("geospatial/js/gps_capture",[
+    "jquery",
+    "knockout",
+    'underscore',
+    'hqwebapp/js/initial_page_data',
+    "hqwebapp/js/bootstrap3/components.ko", // for pagination
+], function (
+    $,
+    ko,
+    _,
+    initialPageData
+) {
+    'use strict';
+
+    var dataItemModel = function (options, dataType) {
+        options = options || {};
+        options = _.defaults(options, {
+            name: '',
+            id: '',  // Can be case_id or user_id
+            lat: '',
+            lon: '',
+        });
+
+        var self = ko.mapping.fromJS(options);
+        self.url = ko.observable();
+        self.dataType = dataType;
+        if (self.dataType === 'user') {
+            self.url(initialPageData.reverse('edit_commcare_user', options.id));
+        } else {
+            self.url(initialPageData.reverse('case_data', options.id));
+        }
+        self.hasUnsavedChanges = ko.observable(false);
+        self.isLatValid = ko.observable(true);
+        self.isLonValid = ko.observable(true);
+
+        self.onMapCaptureStart = function () {
+            // TODO: Implement this function
+        };
+
+        self.onValueChanged = function () {
+            self.hasUnsavedChanges(true);
+
+            self.lat(self.lat().substr(0, 20));
```

Comment:
do we simply truncate the rest?

Reply:
That is correct. If the user is copying these values from an external source then I don't see them ever reaching this limit. For example, a coordinate with 1mm precision is 8 decimal places. Even if a user does insert a decimal over 20 characters, I don't think that level of precision will be necessary and so it should be fine to just automatically truncate the value.

Reply:
Okay, i wonder if we don't truncate at all? I don't understand why we would truncate. And if we are to really truncate then why not truncate at 8 decimal places but 20 total characters.

Reply:
This came in as a [request from QA](https://dimagi-dev.atlassian.net/browse/QA-5585) where they asked if we could restrict the input to 20 characters. Looking at other places in HQ where we accept lat/lon coordinates, such as the "Edit Location" page, the input length is also restricted there. I think a limit is good, as this will prevent users from sending through abnormally large chunks of text.

As an alternative to truncating, we could simply give a validation error if the input is past a certain length (similar to how we handle checking the value) but this would require extra code to manage this validation, and so I think simply truncating the input is fine.